### PR TITLE
tests: benchmarks: Add sysoff with mem retention test.

### PR DESCRIPTION
**Pull request by nordic-babu**
Add tests combining system off and  memory retention with current consumption checking.

### CODE REVIEW

**Review by nordic-pikr — APPROVED** [review]

File: tests/benchmarks/current_consumption/system_off/boards/nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup_ret_mem.overlay [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/ {
```
Comment (nordic-piks) [inline tests/benchmarks/current_consumption/system_off/boards/nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup_ret_mem.overlay:1]: I propose to not mix features into single overlay, rather use nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup.overlay
and add overlay for retention (only that feature in the overlay).
Then at sample.yaml, use both of those overlays.
This also prevents duplication of content of nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup.overlay.

Comment (nordic-babu) [reply tests/benchmarks/current_consumption/system_off/boards/nrf54l15dk_nrf54l15_cpuapp_gpio_wakeup_ret_mem.overlay]: done

**Review by nordic-segl — APPROVED** [review]

File: tests/benchmarks/current_consumption/system_off/CMakeLists.txt [hunk 2]
```diff
@@ -9,3 +9,6 @@ find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(system_off)
 
 target_sources(app PRIVATE $ENV{ZEPHYR_BASE}/samples/boards/nordic/system_off/src/main.c)
+if (CONFIG_APP_USE_RETAINED_MEM)
```
Comment (nordicjm) [inline tests/benchmarks/current_consumption/system_off/CMakeLists.txt:12]:
```suggestion
if(CONFIG_APP_USE_RETAINED_MEM)
```

File: tests/benchmarks/current_consumption/system_off/Kconfig [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+choice
+	prompt "Use retention"
+	optional
+
+config APP_USE_RETAINED_MEM
+	bool "Use state retention in system off using retained_mem driver"
+	depends on RETAINED_MEM
+
+endchoice
```
Comment (nordicjm) [inline tests/benchmarks/current_consumption/system_off/Kconfig:17]: why is this a choice? Just make it a normal Kconfig? Or get rid of it entirely and just check if RETAINED_MEM is enabled, don't see the point in another Kconfig...

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: tests/benchmarks/current_consumption/system_off/Kconfig (hunk 3 above)
Comment (nordic-babu) [reply tests/benchmarks/current_consumption/system_off/Kconfig:17]: it is true that this definition is redundant, removed it

**Review by nordic-piks — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**